### PR TITLE
Fix relatioship fieldset

### PR DESCRIPTION
**Pull request by PhyberApex**
This is the fix for #235 

As mentioned there relationships should also be filtered by the sparse fieldset.

### CODE REVIEW

**Comment by PhyberApex** [comment]
Sorry for all those commits in this again... we were using this internally and we needed the changes from both PRs and I did not know how to handle this better.

~Cheers

**Comment by PhyberApex** [comment]
Any update on this?

**Comment by joukevandermaas** [comment]
I'm sorry, this completely slipped past me during the holidays. I will take a look at this very soon.

File: Tests/JsonApiSerializerTests.cs [hunk 1]
```diff
@@ -83,6 +83,22 @@ public void UsesQueryFieldsetExpressions()
             Assert.NotNull(result["data"][0]["attributes"]["location"]);
             Assert.Null(result["data"][0]["attributes"]["number-of-employees"]);
         }
+        
+        [Fact(DisplayName = "Uses query fieldset expressions also on relationships if specified")]
```
Comment (joukevandermaas) [inline Tests/JsonApiSerializerTests.cs]: I guess the "also" will lose its meaning pretty soon after this is merged :smile:

File: Tests/JsonApiSerializerTests.cs [hunk 2]
```diff
@@ -83,6 +83,22 @@ public void UsesQueryFieldsetExpressions()
...
+            {
+                AllowQuery = true
+            };
+            var people = Get.People(1).ToList();
+            var result = target.Serialize(people, new Uri(DefaultUrl, "?fields[person]=Job"));
+            _output.WriteLine(result.ToString());
+
+            var relationships = result["data"][0]["relationships"];
+
+            Assert.Null(relationships["family-members"]);
```
Comment (joukevandermaas) [inline Tests/JsonApiSerializerTests.cs]: Should you also assert that `Job` actually is there?

File: Tests/JsonApiSerializerTests.cs (hunk 1 above)
Comment (PhyberApex) [reply Tests/JsonApiSerializerTests.cs]: I do not quite get what you mean here, I hope I resolved the issue.

File: Tests/JsonApiSerializerTests.cs (hunk 2 above)
Comment (PhyberApex) [reply Tests/JsonApiSerializerTests.cs]: I should and I will :)

**Comment by joukevandermaas** [comment]
Thanks very much!

File: Tests/JsonApiSerializerTests.cs (hunk 1 above)
Comment (joukevandermaas) [reply Tests/JsonApiSerializerTests.cs]: You did, it wasn't really an issue just a stylistic thing :smile: if you read the test title by itself, the "also" didn't make sense, it only did in the context of the other tests. It was a very minor thing anyway.

Comment (PhyberApex) [reply Tests/JsonApiSerializerTests.cs]: Ahh I see. Didn't get it probably because english istn't my first language :)

**Pull request merged**